src/backbone_router/nd_proxy.cpp

```diff
@@ -0,0 +1,575 @@
+                    Ip6Address &        dst     = *reinterpret_cast<Ip6Address *>(&pktinfo->ipi6_addr);
+                    uint32_t            ifindex = pktinfo->ipi6_ifindex;
+
+                    for (const Ip6Address &ipaddr : mNdProxySet)
+                    {
+                        if (ipaddr.ToSolicitedNodeMulticastAddress() == dst)
+                        {
+                            found = true;
+                            break;
+                        }
+                    }
+
+                    otbrLog(OTBR_LOG_DEBUG, "NdProxyManager: dst=%s, ifindex=%d, proxying=%s", dst.ToString().c_str(),
+                            ifindex, found ? "Y" : "N");
+                }
+                break;
+
+            case IPV6_HOPLIMIT:
+                if (cmsghdr->cmsg_len == CMSG_LEN(sizeof(int)))
+                {
+                    int hops = *(int *)CMSG_DATA(cmsghdr);
+
+                    otbrLog(OTBR_LOG_DEBUG, "NdProxyManager: hops=%d (%s)", hops, hops == 255 ? "Good" : "Bad");
+
+                    VerifyOrExit(hops == 255);
+                }
+                break;
+            }
+        }
+
+        VerifyOrExit(found, error = OTBR_ERROR_NOT_FOUND);
+
+        {
+            struct nd_neighbor_solicit *ns     = reinterpret_cast<struct nd_neighbor_solicit *>(packet);
+            Ip6Address &                target = *reinterpret_cast<Ip6Address *>(&ns->nd_ns_target);
+
+            otbrLog(OTBR_LOG_INFO, "NdProxyManager: send solicited NA for multicast NS: src=%s, target=%s",
+                    src.ToString().c_str(), target.ToString().c_str());
+
+            SendNeighborAdvertisement(target, src);
+        }
+    }
+
+exit:
+    otbrLogResult(error, "NdProxyManager: %s", __FUNCTION__);
+    return;
+}
+
+void NdProxyManager::ProcessUnicastNeighborSolicition(void)
+{
+    otbrError error = OTBR_ERROR_NONE;
+    char      packet[kMaxICMP6PacketSize];
+    ssize_t   len;
+
+    VerifyOrExit((len = recv(mUnicastNsQueueSock, packet, sizeof(packet), 0)) >= 0, error = OTBR_ERROR_ERRNO);
+    VerifyOrExit(nfq_handle_packet(mNfqHandler, packet, len) == 0, error = OTBR_ERROR_ERRNO);
+
+    error = OTBR_ERROR_NONE;
+
+exit:
+    otbrLogResult(error, "NdProxyManager: %s", __FUNCTION__);
+}
+
+void NdProxyManager::HandleBackboneRouterNdProxyEvent(otBackboneRouterNdProxyEvent aEvent, const otIp6Address *aDua)
+{
+    Ip6Address target;
+
+    if (aEvent != OT_BACKBONE_ROUTER_NDPROXY_CLEARED)
+    {
+        assert(aDua != nullptr);
+        target = Ip6Address(aDua->mFields.m8);
+    }
+
+    switch (aEvent)
+    {
+    case OT_BACKBONE_ROUTER_NDPROXY_ADDED:
+    case OT_BACKBONE_ROUTER_NDPROXY_RENEWED:
+    {
+        bool isNewInsert = mNdProxySet.insert(target).second;
+
+        if (isNewInsert)
+        {
+            JoinSolicitedNodeMulticastGroup(target);
+        }
+
+        SendNeighborAdvertisement(target, Ip6Address::GetLinkLocalAllNodesMulticastAddress());
+    }
+    break;
```

Comment:
nit: to maintain alignment of break statements
```suggestion
        break;
    }
```

Reply:
I think I need to put the break in the {} block.